### PR TITLE
Converted task command extension to new pattern

### PR DESCRIPTION
**Pull request by jtpetty**

### CODE REVIEW

File: src/Agent.Worker/WorkerCommandManager.cs [hunk 1]
```diff
@@ -160,20 +160,19 @@ protected void InstallWorkerCommand(IWorkerCommand commandExecutor)
         {
             if (_commands.ContainsKey(commandExecutor.Name))
             {
-                //TODO: Add some logging (or throw exception)
-                return;
+                throw new Exception(StringUtil.Loc("CommandDuplicateDetected", commandExecutor.Name, CommandArea.ToLowerInvariant()));
             }
             _commands[commandExecutor.Name] = commandExecutor;
             var aliasList = commandExecutor.Aliases;
             if (aliasList != null)
```
[the remark this answers is not in the document]
Comment (jtpetty) [reply src/Agent.Worker/WorkerCommandManager.cs]: no problem, I'll do that when I convert the last command extension

File: src/Agent.Worker/WorkerCommandManager.cs [hunk 2]
```diff
@@ -160,20 +160,19 @@ protected void InstallWorkerCommand(IWorkerCommand commandExecutor)
...
             {
                 foreach (var alias in commandExecutor.Aliases)
                 {
-                    if (!_commands.ContainsKey(alias))
+                    if (_commands.ContainsKey(alias))
                     {
-                        _commands[alias] = commandExecutor;
+                        throw new Exception(StringUtil.Loc("CommandDuplicateDetected", alias, CommandArea.ToLowerInvariant()));
                     }
-                    // TODO: else add some logging (or throw exception)
```
Comment (stephenmichaelf) [inline src/Agent.Worker/WorkerCommandManager.cs]: Previously this would fail silently and now we are going to blow. How sure are we that won't happen? :)

Comment (jtpetty) [reply src/Agent.Worker/WorkerCommandManager.cs]: If it blows up, we know we didn't do a conversion correctly and we are glad we caught it before it went out :)

File: src/Test/L0/Worker/WorkerCommandManagerL0.cs [hunk 3]
```diff
@@ -14,9 +15,13 @@ public TestWorkerCommandExtensionL0()
                 CommandArea = "TestL0";
                 SupportedHostTypes = HostTypes.All;
                 InstallWorkerCommand(new FooCommand());
-                InstallWorkerCommand(new Foo2Command());
                 InstallWorkerCommand(new BarCommand());
             }
+
+            public void InstallFoo2Command()
+            {
+                InstallWorkerCommand(new FooCommand());
```
Comment (stephenmichaelf) [inline src/Test/L0/Worker/WorkerCommandManagerL0.cs]: My understanding is this is set up so we throw when installing more than one of the same command.

Would be more clear if we rename this maybe something like `InstallSecondFooCommand`

Comment (jtpetty) [reply src/Test/L0/Worker/WorkerCommandManagerL0.cs]: actually, that was a typo. I meant to install a Foo2Command

**Pull request merged**